packages/mookme/src/utils/git.ts

```diff
@@ -35,21 +35,21 @@ export class GitToolkit {
 
   getNotStagedFiles(): string[] {
     debug(`getNotStagedFiles called`);
-    const notStagedFiles = execSync('git diff --name-only').toString().split('\n');
+    const notStagedFiles = execSync('git diff --name-only --diff-filter=d').toString().split('\n');
```

Comment:
Hi @jfb-workhelix Thanks for trying to fix this really annoying issue! I had started to take a look to and used `--diff-filter=ACMR` instead since we **don't** want to include deleted files, see the [git docs](https://git-scm.com/docs/git-diff#Documentation/git-diff.txt---diff-filterACDMRTUXB82308203).

Reply:
Hello, I might be missing something here but per [Git's diff-filter documentation](https://git-scm.com/docs/git-diff#Documentation/git-diff.txt---diff-filterACDMRTUXB82308203), the lowercased argument means we *exclude* deleted files from the filter.

> Also, these upper-case letters can be downcased to exclude. E.g. --diff-filter=ad excludes added and deleted paths.

This is what we want isn't it ?

Reply:
Haaa! Yeah :D I guess I shouldn't have stopped at the first paragraph 🙄 My bad @jfb-workhelix !

I was able to get a test working but maybe it needs a better setup? It's also probably not covering all use cases but it's a start ¯\_(ツ)_/¯ 

```diff
diff --git a/packages/mookme/tests/cases/pre-commit-succeeds.sh b/packages/mookme/tests/cases/pre-commit-succeeds.sh
index e5f7293..b080d73 100755
--- a/packages/mookme/tests/cases/pre-commit-succeeds.sh
+++ b/packages/mookme/tests/cases/pre-commit-succeeds.sh
@@ -9,10 +9,13 @@ npm run build &> /dev/null
 cd $TESTS_DIR

 git init
+git commit --allow-empty -m "first commit"
 mkdir -p package1
+mkdir -p package1/.hooks
 mkdir -p parent1/package2
 mkdir -p parent1/package3
 mkdir -p parent1/package3/.hooks
+touch package1/tobedeleted.txt

 node $ROOT_FOLDER/dist/index.js init \
     --yes \
@@ -26,7 +29,13 @@ node $ROOT_FOLDER/dist/index.js init \
 chmod +x .hooks/partials/get-dir-name

 echo '{"steps": [{"name": "Hello world !", "command": "echo 'hello'"}]}' > .hooks/pre-commit.json
+echo '{"type": "txt", "steps": [{"name": "Ignore deleted files", "command": "cat {matchedFiles}"}]}' > package1/.hooks/pre-commit.json
 echo '{"steps": [{"name": "Hello world !", "command": "[  $(get-dir-name) = 'package3' ]"}]}' > parent1/package3/.hooks/pre-commit.json

 git add .
-node $ROOT_FOLDER/dist/index.js run -t pre-commit
\ No newline at end of file
+node $ROOT_FOLDER/dist/index.js run -t pre-commit
+
+git commit -m "test: commit" --no-verify
+git rm package1/tobedeleted.txt
+
+node $ROOT_FOLDER/dist/index.js run -t pre-commit
```

Without your patch, we would get the following error:

```bash
Hook of package package1 failed at step Ignore deleted files

cat: tobedeleted.txt: No such file or directory
```